composer/callbacks/memory_monitor.py

```diff
@@ -63,8 +63,13 @@ def get_memory_report() -> Dict[str, Union[int, float]]:
     if not torch.cuda.is_available():
         log.debug("Cuda is not available. The memory report will be empty.")
         return {}
-    device_stats = torch.cuda.memory_stats()
-    memory_report = {}
-    for torch_stat_name, stat_alias in _MEMORY_STATS.items():
-        memory_report[stat_alias] = device_stats[torch_stat_name]
+    memory_stats = torch.cuda.memory_stats()
+
+    if len(memory_stats) == 0:
+        log.debug("No GPU memory was used, returning empty.")
```

Comment:
Could we raise an exception rather than return an empty dict if no gpu is available? An empty dict will clutter the logs.

The memory monitor should only be used if on GPU.

Reply:
We do raise an exception if no GPU is available (see a few lines above). This is for the case where a GPU is available, but since no GPU memory was allocated, the memory stats is empty.

Reply:
Ahhh ok, my bad. In this case, then I think it would make sense to return `None` and make any caller code handle missing memory stats, but will create a separate issue to clean this up later....